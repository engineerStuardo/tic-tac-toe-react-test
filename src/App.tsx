import Board from './components/Board';
import Moves from './components/Moves';
import { useCalculation } from './hooks/useCalculation';

function App() {
  const { history, stepNumber, handleClick, jumpTo, winner, xO } =
    useCalculation();

  return (
    <>
      <h1 data-testid='title'>Tic Tac Toe - React Test</h1>
      <Board squares={history[stepNumber]} onClick={handleClick} />
      <div data-testid='info-wrapper' className='info-wrapper'>
        <div>
          <h3>History</h3>
          <Moves history={history} jumpTo={jumpTo} />
        </div>
        <h3>{winner ? 'Winner: ' + winner : 'Next Player: ' + xO}</h3>
      </div>
    </>
  );
}

export default App;
