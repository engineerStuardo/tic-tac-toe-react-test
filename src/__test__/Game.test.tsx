import React from 'react';
import { render, screen } from '@testing-library/react';

import App from '../App';

describe('Game component', () => {
  test('Should match a snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test('Should display title in h1 tag', () => {
    render(<App />);
    expect(screen.getByTestId('title')).toBeTruthy();
  });

  test('Title should be "React Tic Tac Toe - With Hooks"', () => {
    render(<App />);
    expect(screen.getByText('Tic Tac Toe - React Test')).toBeTruthy();
  });

  test('Should have one "info-wrapper" class', () => {
    render(<App />);
    expect(screen.getByTestId('info-wrapper')).toBeTruthy();
  });

  test('"History" text should be in the screen', () => {
    render(<App />);
    expect(screen.getByText('History')).toBeTruthy();
  });

  test('"Next Player: X" text should be in the screen when the game starts', () => {
    render(<App />);
    expect(screen.getByText('Next Player: X')).toBeTruthy();
  });
});
