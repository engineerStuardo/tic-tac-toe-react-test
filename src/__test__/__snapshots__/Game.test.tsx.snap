// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component Should match a snapshot 1`] = `
<div>
  <h1
    data-testid="title"
  >
    Tic Tac Toe - React Test
  </h1>
  <div
    class="board"
    data-testid="board"
  >
    <button
      class="squares"
    />
    <button
      class="squares"
    />
    <button
      class="squares"
    />
    <button
      class="squares"
    />
    <button
      class="squares"
    />
    <button
      class="squares"
    />
    <button
      class="squares"
    />
    <button
      class="squares"
    />
    <button
      class="squares"
    />
  </div>
  <div
    class="info-wrapper"
    data-testid="info-wrapper"
  >
    <div>
      <h3>
        History
      </h3>
      <li>
        <button>
          Go to Start
        </button>
      </li>
    </div>
    <h3>
      Next Player: X
    </h3>
  </div>
</div>
`;
